rs/registry/canister/src/mutations/do_swap_node_in_subnet_directly.rs

```diff
@@ -27,14 +33,64 @@ impl Registry {
 
         // Check if the payload is valid by itself.
         payload.validate()?;
+        let (old_node_id, _new_node_id) =
```

Comment:
nit: Use destructuring here (i.e.
```
let SwapNodeInSubnetDirectlyPayload { old_node_id, new_node_id, ...otherfields...} = payload;
```

Reply:
As far as I know that doesn't work if `old_node_id` is `Option<T>`. I may have misunderstood your idea but I can't do something like:
```rust
let SwapNodeInSubnetDirectlyPayload {
     old_node_id: Some(old_node_id),
     new_node_id: Some(new_node_id),
} = payload;
```

Did I misunderstand what you wanted to say?